src/expr/src/expr/expr_array_length.rs

```diff
@@ -0,0 +1,278 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_common::array::{ArrayRef, DataChunk};
+use risingwave_common::row::OwnedRow;
+use risingwave_common::types::{DataType, Datum, DatumRef, ScalarImpl, ScalarRefImpl, ToDatumRef};
+use risingwave_common::util::iter_util::ZipEqFast;
+use risingwave_pb::expr::expr_node::{RexNode, Type};
+use risingwave_pb::expr::ExprNode;
+
+use crate::expr::{build_from_prost, BoxedExpression, Expression};
+use crate::{bail, ensure, ExprError, Result};
+
+/// Returns the length of an array.
+///
+/// ```sql
+/// array_length ( array anyarray) → int64
+/// ```
+///
+/// Examples:
+///
+/// ```slt
+/// query T
+/// select array_length(null::int[]);
+/// ----
+/// NULL
+///
+/// query T
+/// select array_length(array[1,2,3]);
+/// ----
+/// 3
+///
+/// query T
+/// select array_length(array[1,2,3,4,1]);
+/// ----
+/// 5
+///
+/// query T
+/// select array_length(null::int[]);
+/// ----
+/// NULL
+///
+/// query T
+/// select array_length([[1, 2, 3]]);
+/// ----
+/// 1
+///
+/// /// query T
```

Comment:
```suggestion
/// query T
```

CI failed here

Reply:
Oh okay, I will fix this. I thought the CI failed because I did not compile the dashboard code.